src/main/java/org/openrewrite/java/testing/junit5/AddParameterizedTestAnnotation.java

```diff
@@ -0,0 +1,103 @@
+                // must be a better way to writer this
+                return (VALUE_SOURCE_MATCHER.matches(ann) || CSV_SOURCE_MATCHER.matches(ann) ||
+                        METHOD_SOURCE_MATCHER.matches(ann) || NULL_SOURCE_MATCHER.matches(ann) ||
+                        EMPTY_SOURCE_MATCHER.matches(ann) || NULL_EMPTY_SOURCE_MATCHER.matches(ann) ||
+                        ENUM_SOURCE_MATCHER.matches(ann) || CSV_FILE_SOURCE_MATCHER.matches(ann) ||
+                        ARG_SOURCE_MATCHER.matches(ann));
+            }
+
+            public List<J.Annotation> reorderAnnotations(List<J.Annotation> annotations) {
+                // if @ParameterizedTest is already at start then return early
+                if (PARAM_TEST_MATCHER.matches(annotations.get(0))) { return annotations; }
+                List<J.Annotation> ordered = new ArrayList<>(annotations);
+
+                // get @ParameterizedTest
+                List<J.Annotation> filteredList = ordered.stream().filter(PARAM_TEST_MATCHER::matches).collect(Collectors.toList());
+                J.Annotation paramAnn = filteredList.get(0);
+
+                ordered.remove(paramAnn);
+                ordered.add(0, paramAnn);
+
+                return ordered;
+            }
+
+            @Override
+            public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration md, ExecutionContext ctx) {
+                J.MethodDeclaration m = super.visitMethodDeclaration(md, ctx);
+
+                // return early if @ValueSource and siblings are not detected
+                if (m.getLeadingAnnotations().stream().noneMatch(this::checkForSourceAnnotation) ||
+                    m.getLeadingAnnotations().stream().anyMatch(PARAM_TEST_MATCHER::matches)) {
+                    return m;
+                }
+
+                JavaCoordinates coordinates = m.getCoordinates().addAnnotation(Comparator.comparing(J.Annotation::getSimpleName));
+
+                for (J.Annotation ann : m.getLeadingAnnotations()) {
+                    if (TEST_ANNOTATION_MATCHER.matches(ann)) {
+                        coordinates = ann.getCoordinates().replace();
+                        break;
+                    }
+                }
+
+                m = JavaTemplate.builder("@ParameterizedTest")
+                        .javaParser(JavaParser.fromJavaVersion()
+                                .classpathFromResources(ctx, "junit-jupiter-api-5.9", "junit-jupiter-params-5.9"))
+                        .imports("org.junit.jupiter.params.ParameterizedTest")
+                        .build()
+                        .apply(getCursor(), coordinates);
+                maybeAddImport("org.junit.jupiter.params.ParameterizedTest");
+                maybeRemoveImport("org.junit.jupiter.api.Test");
+                List<J.Annotation> ordered = reorderAnnotations(m.getLeadingAnnotations());
+
+                return m.withLeadingAnnotations(ordered);
```

Comment:
This ought to help your formatting issues, as long as your consistent in your before and after templates in the tests.
```suggestion
                return maybeAutoFormat(m , m.withLeadingAnnotations(ordered), ctx);
```

Reply:
Using `maybeAutoFormat()` helped but the `replaceTestWithParameterizedTestRegardlessOfOrder()`is still failing for formatting issues. Its now adding a newline above the method and two new spaces at the start of each line that the method is on. Seems like the only thing different about this test from the others is that the `reorderAnnotations()` method runs for this test. Could the issue be there maybe?

Reply:
You're also somewhat inconsistent with whitespace in your samples; [see below](https://github.com/openrewrite/rewrite-testing-frameworks/pull/353#discussion_r1229268796). I guess clearing that up should help your tests to pass or get closer.

Reply:
I saw that comment as well and I made sure to go through all the samples and makes sure they all had the same indentation and spacing. Its just this one that is still outputting this weird format. I've also tried using cmd+shift+L on all the samples to see if maybe that would format something I was missing.